### PR TITLE
doc: fix broken links, wrong artifactType and outdated ORAS usage

### PR DESCRIPTION
**Pull request by yizha1**
Updates:
- Fix broken links
- Fix wrong artifactType
- Removing ORAS information

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

File: specs/signing-and-verification-workflow.md [hunk 1]
```diff
@@ -38,15 +37,14 @@ The user wants to pull an OCI artifact only if they are signed by a trusted publ
...
+1. **Should Notary v2 verify the signature? :** Depending upon [trust-policy](./trust-store-trust-policy.md#trust-policy) configuration, determine whether Notary v2 needs to verify the signature or not.
    If signature verification should be skipped for the given artifact, skip the below steps and directly jump to step 4.
-1. **Get signature artifact descriptors:** Using the ORAS [Manifest Referrers API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) download the Notary v2 signature artifact descriptors.
-   The  `artifactType` parameter is set to Notary v2 signature's artifact type `application/vnd.cncf.notary.v2.signature`.
-   Note that all ORAS implementations may not support filtering using `artifactType`.
+1. **Get signature artifact descriptors:** Using the [OCI Distribution Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) download the Notary v2 signature artifact descriptors.
+   The  `artifactType` parameter is set to Notary v2 signature's artifact type `application/vnd.cncf.notary.signature`.
```
Comment (shizhMSFT) [inline specs/signing-and-verification-workflow.md]: nit:
```suggestion
   The `artifactType` parameter is set to Notary v2 signature's artifact type `application/vnd.cncf.notary.signature`.
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Comment by dasiths** [comment]
Just mentioning a couple of important changes here so that search indexing will pick it up.

`mediaType` parameter changed from `application/vnd.cncf.oras.artifact.manifest.v1+json` to `application/vnd.oci.artifact.manifest.v1+json`

and 

`artifactType` parameter changed from `application/vnd.cncf.notary.v2.signature` to `application/vnd.cncf.notary.signature`

**Pull request merged**